### ISSUE TITLE
Refresh identifiers names.

### ISSUE THREAD

**Issue by iago-lito** [BecksLab/EcologicalNetworksDynamics.jl#114]
Before release, all identifiers in the project (or at least all `export`ed identifiers) should be screened for consistency, clarity, confusability, simplicity, conciseness *etc.*. Let's start discussing this now, I'm slacking the whole list here just to start discussion:

```
                   ModelParameters → Model
                    get_parameters → get_model
                           FoodWeb ✓
                  MultiplexNetwork ✓

                      (B/dB/dBdt!) → (u/du/dudt!) (not exported)
                     generate_dbdt → generate_dudt

                          BioRates | These ones are just two loose bags with
                       Environment | some model parameters inside, unclear?
                                   | => Drop them after #84.

                FunctionalResponse ✓
                    LinearResponse ✓
                   ClassicResponse ✓
              BioenergeticResponse ✓

    BioEnergeticFunctionalResponse ⚠ Not defined within the project! => Remove.

               TemperatureResponse → TemperatureScale
             NoTemperatureResponse → NoTemperatureScale
                     ExponentialBA → ExponentialBoltzmannArrheniusScale
                  set_temperature! → set_temperature_scale!

                  AllometricParams → Allometry (see #74 for possible evolution)
               ExponentialBAParams → ExponentialBoltzmannArrheniusAllometry (see #74)
                   allometric_rate → get_rates / get_allometric_rates ?

               DefaultGrowthParams → default_growth
       DefaultMaxConsumptionParams → default_max_consumption
           DefaultMetabolismParams → default_metabolism / default_loss ?
            DefaultMortalityParams → default_mortality / default_death ?

               get_extinct_species → get_extinctions     | see #110
                                ∅  → get_extinct_species |

                  species_richness → richness | Shorten.
               species_persistence            | Keep as-is or it'd be confusing ?

                exp_ba_attack_rate |
          exp_ba_carrying_capacity | Expanding these to full names would be not
                     exp_ba_growth | satisfying either because they would be
              exp_ba_handling_time | very, very long. Maybe namespace them?
                exp_ba_matrix_rate |
                 exp_ba_metabolism | Maybe something along the following instead?
              exp_ba_params_to_vec |   attack_rate(:EBA, ...)
                exp_ba_vector_rate | But see #74.

                A_competition_full | Does 'full' actually means 'dense' here?
               A_facilitation_full | Maybe a `get_dense_matrix(:competition)`
               A_interference_full | would be a simpler option?
                     A_refuge_full |

                      cascademodel | re-exported from `EcologicalNetworks.jl`,
                        nichemodel | not much we can do about them
              nestedhierarchymodel | unless we are willing to alias them?
                          mpnmodel | (not a *very* good idea: confusion ahead)

                              cpad | Out of scope: keep but don't export.
                             fitin | This is my bad: sorry ^ ^"

                    @check_between |
               @check_greater_than | Do these really need to be exported?
                         @check_in | They feel out-of-scope.
                 @check_lower_than | => Stop exporting them.
                       @check_size |

                        foodweb_cv |
                  foodweb_evenness |
                  foodweb_richness | About to change with #87 anyway.
                   foodweb_shannon |
                   foodweb_simpson |

                       attack_rate |
                         boltzmann |
          coefficient_of_variation | Nothing much to say yet. Any thoughts?
                       connectance |
             draw_asymmetric_links |
              draw_symmetric_links |
                        efficiency |
                ExtinctionCallback |
                 find_steady_state |
                     handling_time |
            homogeneous_preference |
                 interaction_names |
                             Layer |
multiplex_network_parameters_names |
                           n_links |
       nontrophic_adjacency_matrix |
               NonTrophicIntensity |
              population_stability |
       potential_competition_links |
      potential_facilitation_links |
      potential_interference_links |
            potential_refuge_links |
                      predators_of |
                          preys_of |
                   producer_growth |
               ProducerCompetition |
                         producers |
                          richness |
                          simulate |
                     total_biomass |
                    trophic_levels |
```

Also I can think of these few non-exported identifiers for now:
```
                effect_competition → competition_effect
               effect_facilitation → facilitation_effect
                     effect_refuge → refuge_effect
```
Any other idea?

Regarding the files under `src/` we could also take this opportunity to sort'em out a bit. In particular, the distinction between the three folders "before simulation / during simulation / after simulation" (IIUC) needs to be made clearer:

```
src/EcologicalNetworksDynamics.jl          ✓
src/formatting.jl                          ✓
src/macros.jl                              ✓

src/inputs/                                ✓ or src/model/ or src/parameters/ : what we feed the simulation with)
src/model/                                 → src/simulate/
src/measures/                              ✓or src/measure/

src/utils.jl                               → src/(input|model|parameters)/utils.jl?
src/aliasing_dict.jl                       → src/(input|model|parameters)/aliasing_dict.jl?

src/inputs/temperature_dependent_rates.jl  ✓
src/inputs/foodwebs.jl                     ✓
src/inputs/nontrophic_interactions.jl      ✓
src/inputs/producer_competition.jl         ✓

src/inputs/environment.jl                  | Float a little, how are they consistent?
src/inputs/biological_rates.jl             |

src/inputs/MultiplexNetwork_signature.jl   → multiplex_network_signature.jl

src/inputs/temperature_size_rule.jl        ⚠ Empty file!

src/inputs/functional_response.jl          ⚠ Mix of "before simulation"- and "after simulation"-related code.

src/measures/utils.jl                      |
src/measures/structure.jl                  | About to change in #87.
src/measures/functioning.jl                |
src/measures/stability.jl                  |

src/model/set_temperature.jl               → src/(inputs|model|parameters)/set_temperature.jl
src/model/model_parameters.jl              → src/(inputs|model|parameters)/model.jl

src/model/consumption.jl                   ✓
src/model/productivity.jl                  ✓
src/model/metabolic_loss.jl                ✓
src/model/effect_nti.jl                    → nti_effects.jl
src/model/dbdt.jl                          → dudt.jl
src/model/generate_dbdt.jl                 → src/simulate/generate_dudt/main.jl
src/model/generate_dbdt_raw.jl             → src/simulate/generate_dudt/raw.jl
src/model/generate_dbdt_compact.jl         → src/simulate/generate_dudt/compact.jl
src/model/simulate.jl                      ✓or src/simulate/main.jl
```

Also, the files under the `test/` folder do not need to be all prefixed by `test-`, or to match the files under `src` exactly, as long as they all live under `test/` and all features are tested. So what about the following?

```
test/runtests.jl                                ✓
test/test-utils.jl                              ✓ ./test/utils.jl
test/measures/test-utils.jl                     → ./test/measures_utils.jl

test/inputs/test-producer_competition.jl        → ./test/producer_competition.jl
test/inputs/test-biological_rates.jl            → ./test/biological_rates.jl
test/inputs/test-temperature_dependent_rates.jl → ./test/temperature_dependent_rates.jl
test/inputs/test-foodwebs.jl                    → ./test/foodwebs.jl
test/inputs/test-environment.jl                 → ./test/environment.jl
test/inputs/test-nontrophic_interactions.jl     → ./test/nontrophic_interactions.jl
test/inputs/test-functional_response.jl         → ./test/functional_response.jl
test/measures/test-stability.jl                 → ./test/stability.jl
test/measures/test-functioning.jl               → ./test/functioning.jl
test/measures/test-structure.jl                 → ./test/structure.jl
test/model/test-nutrientintake.jl               → ./test/nutrientintake.jl
test/model/test-zombies.jl                      → ./test/zombies.jl
test/model/test-model_parameters.jl             → ./test/model_parameters.jl
test/model/test-productivity.jl                 → ./test/productivity.jl
test/model/test-effect_nti.jl                   → ./test/effect_nti.jl
test/model/test-metabolic_loss.jl               → ./test/metabolic_loss.jl
test/model/test-simulate.jl                     → ./test/simulate.jl
test/model/test-consumption.jl                  → ./test/consumption.jl
test/model/test-set_temperature.jl              → ./test/set_temperature.jl
test/test-basic-pipeline.jl                     → ./test/basic-pipeline.jl
```

Parameter names should also be checked, see #2 for example.

What do you think ? :)
